src/client/consumer.ts

```diff
@@ -70,28 +73,40 @@ class Consumer implements vscode.Disposable {
      * Received messages and/or errors are emitted via events.
      */
     async start(): Promise<void> {
+        const partition = this.options.partition;
+        const fromOffset = this.options.fromOffset;
+        const topic = this.options.topic;
         this.kafkaClient = createKafka(this.options);
-        this.consumer = this.kafkaClient.consumer({ groupId: `vscode-kafka-${this.options.clusterId}-${this.options.topic}`, retry: { retries: 3 }});
+        this.consumer = this.kafkaClient.consumer({ groupId: this.options.consumerGroupId, retry: { retries: 3 } });
         await this.consumer.connect();
-        await this.consumer.subscribe({ topic: this.options.topic, fromBeginning: this.options.fromOffset ===  "earliest" })
+
+        const fromBeginning = fromOffset === "earliest" ? true : (fromOffset === "latest" ? false : undefined);
+        const subscribeOptions = (fromBeginning != undefined) ? { topic, fromBeginning } : { topic };
+        await this.consumer.subscribe(subscribeOptions)
 
         await this.consumer.run({
-            eachMessage: async ({ topic , partition, message }) => {
+            eachMessage: async ({ topic, partition, message }) => {
                 this.onDidReceiveMessageEmitter.fire({
                     uri: this.uri,
                     record: { topic: topic, partition: partition, ...message },
                 });
             },
         });
+
+        if (partition || (fromOffset && fromBeginning == undefined)) {
+            this.consumer.seek({ topic, partition: partition ? partition : 0, offset: fromOffset ? fromOffset : '0' });
+        }
+
     }
 
-    private parseUri(uri: vscode.Uri): { clusterId: string; topic: string; partition?: string} {
-        const [clusterId, topic, partition] = uri.path.split("/");
+    private parseUri(uri: vscode.Uri): { clusterId: string; topic: string; from?: string, partition?: string } {
+        const [clusterId, topic, from, partition] = uri.path.split("/");
```

Comment:
Makes more sense for parameters to be read from query parameters ?from=foo&partition=bar&...

Reply:
Good catch! Fixed.